File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.go
func longestConsecutive(nums []int) int {
    set := make(map[int]bool)

    for _, n := range nums {
        set[n] = true
    }

    long := 0

    for num := range set {
        if !set[num-1] {
            length := 1

            for set[num+length] {
                length++
            }

            if length > long {
                long = length
            }
        }
    }
    return long
}

File: Data Structures & Algorithms/missing-number/submission-0.rs
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let n = nums.len() as i32;

        let mut xor = 0;

        for i in 1..=n {
            xor ^= i
        }

        for num in nums {
            xor ^= num;
        }
        
        xor
    }
}
